### PR TITLE
Hds 1401 header component

### PR DESCRIPTION
**Pull request by harriplappalainen**
## Description

Header component to act as a wrapper for upcoming new navigation components. I imagine this will gain more meat around the bones as child components come to life but this component mainly serves as means to pass on information relevant to the context. As of now it only provides information whether view is in mobile.

I removed everything else that didn't seem sense to keep here when looking at the designs. I thought about Header accepting `theme` property but it might make more sense to give them straight to the subcomponents, as some of them are very specific. 

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1401

## How Has This Been Tested?
Locally on dev machine
[Demo](https://city-of-helsinki.github.io/hds-demo/header/?path=/story/components-header--example)

### CODE REVIEW

File: packages/react/src/components/header/Header.test.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { render } from '@testing-library/react';
+import { axe } from 'jest-axe';
+
+import { Header } from './Header';
+
+describe('<Header /> spec', () => {
+  it('renders the component', () => {
+    const { asFragment } = render(<Header />);
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (minevala) [inline packages/react/src/components/header/Header.test.tsx:11]: new line?

File: packages/react/src/components/header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+   */
+  id?: string;
+}>;
+
+export const Header = ({ children, className }: HeaderProps) => {
+  const isMobile = useMobile();
+  const context: HeaderContextProps = { isMobile };
+  return (
+    <HeaderContext.Provider value={context}>
+      <div className={classNames(styles.header, className)}>
```
Comment (minevala) [inline packages/react/src/components/header/Header.tsx:26]: Should this be a header element?

**Review by minevala — COMMENTED** [review]
A nice start 👍 
A couple of small comments

**Comment by minevala** [comment]
Snapshot test seems to fail

File: packages/react/src/components/header/Header.tsx (hunk 2 above)
Comment (harriplappalainen) [reply packages/react/src/components/header/Header.tsx:26]: Yes! Totally forgot.

**Comment by harriplappalainen** [comment]
Good remarks! I changed the code a bit and added the snapshot tests which I forgot to do initially.

**Review by minevala — APPROVED** [review]
A nice start 🏅

**Pull request merged**